components/tours/hooks.ts

```diff
@@ -217,3 +217,33 @@ export const useHandleNavigationAndExtraActions = (tourCategory: string) => {
         nextStepActions(step);
     }, [nextStepActions, lastStepActions]);
 };
+
+export const useGetTourtipRedraw = (elementId: string) => {
```

Comment:
Nice solution! I think we can simplify this hook a bit and increase its performance while we're at it: getting a box's element width is one of the things that [causes layout/reflow](https://gist.github.com/paulirish/5d52fb081b3570c81e3a?permalink_comment_id=2381287#getting-box-metrics). But resize observer already gets the measured dom ret for us.

I think we can simplify this hook to something like:

```ts
export const useFollowElementDimensions = (elementId: string): DOMRectReadOnly => {
    const [dimensions, setDimensions] = useState(new DOMRect());

    useEffect(() => {
        const element = document.getElementById(elementId);
        if (!element) {
            return undefined;
        }
        const observer = new ResizeObserver((entries) => {
            if (entries.length > 0) {
                setDimensions(entries[0].contentRect);
            }
        });
        observer.observe(element);

        return () => {
            observer.unobserve(element);
        };
    }, [elementId]);

    return dimensions;
}
```

This would make it a very generic hook we can use elsewhere too. And the dep can either pass the domrect directly as a dep, or just pass its `.width`.

Reply:
@pvev it would be great to put inside, [packages/components/src/common](https://github.com/mattermost/mattermost-webapp/tree/master/packages/components/src/common/hooks)